extensions/amp-video/0.1/amp-video.js

```diff
@@ -259,11 +260,20 @@ export class AmpVideo extends AMP.BaseElement {
     // Cached so mediapool operations (eg: swapping sources) don't interfere with this bool.
     this.hasBitrateSources_ =
       !!this.element.querySelector('source[data-bitrate]') ||
-      this.hasAnyCachedSources_();
+      this.hasAnyCachedSources_() ||
+      this.element.hasAttribute('enable-google-video-cache');
 
     installVideoManagerForDoc(element);
 
     Services.videoManagerForDoc(element).register(this);
+
+    // Add cache sources if opted-in and it's not a cached doc already.
+    if (
+      this.element.hasAttribute('enable-google-video-cache') &&
+      !this.element.querySelector('source[amp-orig-src]')
+    ) {
+      return addCacheSources(this);
```

Comment:
Passing entire instances like this is an anti pattern, please pass `win` and `element` if needed.